ush/python/pygfs/utils/marine_da_utils.py

```diff
@@ -74,3 +74,95 @@ def stage_ens_mem(task_config: AttrDict) -> None:
+    ncf.close()
+    logger.info(f"*** history file date: {hist_date} expected date: {ref_date}")
+    assert hist_date == ref_date, 'Inconsistent bkg date'
+
+
+@logit(logger)
+def gen_bkg_list(bkg_path: str, window_begin=' ', yaml_name='bkg.yaml', ice_rst=False) -> None:
+    """
+    Generate a YAML of the list of backgrounds for the pseudo model
+    """
+
+    # Pseudo model parameters (time step, start date)
+    # TODO: make this a parameter
+    dt_pseudo = 3
+    bkg_date = window_begin
+
+    # Construct list of background file names
+    cyc = str(os.getenv('cyc')).zfill(2)
+    gcyc = str((int(cyc) - 6) % 24).zfill(2)  # previous cycle
+    fcst_hrs = list(range(6, 10, dt_pseudo))
+    files = []
+    for fcst_hr in fcst_hrs:
+        files.append(os.path.join(bkg_path, f"ocean.bkg.f{str(fcst_hr).zfill(3)}.nc"))
+
+    # Identify the ocean background that will be used for the  vertical coordinate remapping
+    ocn_filename_ic = './INPUT/MOM.res.nc'
+    test_hist_date(ocn_filename_ic, bkg_date)  # assert date of the history file is correct
+
+    # Copy/process backgrounds and generate background yaml list
+    bkg_list = []
+    for bkg in files:
+        logger.info(f"****************** bkg: {bkg}")
+        # assert validity of the ocean bkg date, remove basename
+        bkg_date = bkg_date + timedelta(hours=dt_pseudo)
+        test_hist_date(bkg, bkg_date)
+        ocn_filename = os.path.splitext(os.path.basename(bkg))[0] + '.nc'
+
+        # prepare the seaice background, aggregate if the backgrounds are CICE restarts
+        ice_filename = ocn_filename.replace("ocean", "ice")
+
+        # prepare list of ocean and ice bkg to be copied to RUNDIR
+        bkg_dict = {'date': bkg_date.strftime('%Y-%m-%dT%H:%M:%SZ'),
+                    'basename': './bkg/',
+                    'ocn_filename': ocn_filename,
+                    'ice_filename': ice_filename,
+                    'read_from_file': 1}
+
+        bkg_list.append(bkg_dict)
+
+    # save pseudo model yaml configuration
+    f = open(yaml_name, 'w')
+    yaml.dump(bkg_list, f, sort_keys=False, default_flow_style=False)
+
+
+@logit(logger)
+def clean_empty_obsspaces(config, target, app='var'):
+    """
+    Remove obs spaces that point to non-existent file and save
+    """
+
+    # obs space dictionary depth is dependent on the application
+    if app == 'var':
+        obs_spaces = config['cost function']['observations']['observers']
+    else:
+        logger.error(f"Error: {app} case not implemented yet!")
+
+    # remove obs spaces that point to a non existant file
+    cleaned_obs_spaces = []
+    for obs_space in obs_spaces:
+        fname = obs_space['obs space']['obsdatain']['engine']['obsfile']
+        if os.path.isfile(fname):
+            cleaned_obs_spaces.append(obs_space)
+        else:
+            logger.info(f"{fname} does not exist, removing obs space")
+            print("File does not exist")
```

Comment:
Is this a fatal error or just a warning?

Reply:
warning/info, will change the logger